tsconfig.json

```diff
@@ -2,24 +2,31 @@
 	"compilerOptions": {
 		"allowJs": true,
 		"checkJs": true,
-		"noEmit": true,
-		"module": "es2015",
+		"emitDeclarationOnly": true,
+		"declaration": true,
+		"module": "esnext",
 		"moduleResolution": "node",
 		"target": "es5",
-		"baseUrl": "packages",
-		"paths": {
-			"@tannin/*": [ "*/index.js" ],
-			"tannin": [ "tannin/index.js" ]
-		}
+		"composite": true,
+
+		"strict": true,
+		"strictNullChecks": true,
+		"noImplicitAny": true,
+		"noUnusedLocals": true,
+		"noUnusedParameters": true,
+		"noImplicitReturns": true,
+		"noFallthroughCasesInSwitch": true,
+
+		/* This needs to be false so our types are possible to consume without setting this */
+		"esModuleInterop": false,
+		"isolatedModules": true
 	},
-	"include": [
-		"packages"
-	],
+	"files": [],
 	"exclude": [
 		"**/build",
 		"**/dist",
 		"**/__tests__",
 		"**/benchmark",
-		"node_modules"
+		"**/node_modules"
 	]
```

Comment:
If in this new setup we rely on each package to define its own files (at least it's own entrypoint?), then aren't both of these configuration values unnecessary?